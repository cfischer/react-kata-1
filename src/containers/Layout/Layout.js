import React from 'react';
import {
  Container,
  Dropdown,
  Header,
  Menu,
  Segment,
  List
} from 'semantic-ui-react';
import { Link } from 'react-router-dom';

function MainHeader() {
  return (
    <Menu fixed='top' inverted>
      <Container>
        <Menu.Item as={Link} to='/' header>
          React Kata
        </Menu.Item>
        <Menu.Item as={Link} to='/'>Home</Menu.Item>
        <Dropdown item simple text='Options'>
          <Dropdown.Menu>
            <Dropdown.Item as={Link} to='/'>All Books and Magazines</Dropdown.Item>
            <Dropdown.Divider />
            <Dropdown.Header>Search</Dropdown.Header>
            <Dropdown.Item as={Link} to='/isbn'>Find by ISBN</Dropdown.Item>
            <Dropdown.Item as={Link} to='/email'>Find by author`s email</Dropdown.Item>
            <Dropdown.Divider />
            <Dropdown.Header>Sorted Content</Dropdown.Header>
            <Dropdown.Item as={Link} to='/sorted'>Sorted by Title</Dropdown.Item>
          </Dropdown.Menu>
        </Dropdown>
      </Container>
    </Menu>
  )
}

function Footer() {
  return (
    <Segment inverted vertical>
      <Container>
        <Header as='h2' inverted>Summary of Instructions</Header>
        <List as='ol' inverted>
          <List.Item as='li'>Your software should read all data from the given CSV files in a meaningful structure.</List.Item>
          <List.Item as='li'>Print out all books and magazines with all their details (with a meaningful output format).</List.Item>
          <List.Item as='li'>Find a book or magazine by its `isbn`.</List.Item>
          <List.Item as='li'>Find all books and magazines by their `authors`â€™ email..</List.Item>
          <List.Item as='li'>Print out all books and magazines with all their details sorted by `title`. This sort should be done for books and magazines together.</List.Item>
        </List>
        <p>
          Please read full instructions on the <a rel="noopener noreferrer" target="_blank" href="https://github.com/echocat/react-kata-1/blob/master/README.md"> README </a> file
        </p>
      </Container>
    </Segment>
  );
}

function Layout({children}) {
  return (
    <>
      <MainHeader />
      <Container style={{ marginTop: '5em' }}>
        <Header as='h1'>React Kata</Header>
          <Segment vertical>
            {children}
          </Segment>
      </Container>
      <Footer />
    </>
  );
}

export default Layout;
