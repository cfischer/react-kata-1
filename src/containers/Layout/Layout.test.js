import React from 'react';
import ReactDOM from 'react-dom';
import Layout from './Layout';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from 'enzyme';


function LayoutWrapper() {
  return (
    <Router>
      <Layout />
    </Router>
  );
}

it('renders Layout', () => {
  const div = document.createElement('div');
  ReactDOM.render(<LayoutWrapper />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('should render correctly with no props', () => {
  const component = render(<LayoutWrapper/>);
  
  expect(component).toMatchSnapshot();
});
