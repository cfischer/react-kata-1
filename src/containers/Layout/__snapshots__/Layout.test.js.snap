// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with no props 1`] = `
Array [
  <div
    class="ui inverted top fixed menu"
  >
    <div
      class="ui container"
    >
      <a
        class="header item"
        href="/"
      >
        React Kata
      </a>
      <a
        class="item"
        href="/"
      >
        Home
      </a>
      <div
        aria-expanded="false"
        class="ui item simple dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-atomic="true"
          aria-live="polite"
          class="text"
          role="alert"
        >
          Options
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
        >
          <a
            class="item"
            href="/read"
            role="option"
          >
            Output from data read
          </a>
          <a
            class="item"
            href="/all"
            role="option"
          >
            All Books and Magazines
          </a>
          <div
            class="divider"
          />
          <div
            class="header"
          >
            Search
          </div>
          <a
            class="item"
            href="/isbn/"
            role="option"
          >
            Find by ISBN
          </a>
          <a
            class="item"
            href="/email/"
            role="option"
          >
            Find by author\`s email
          </a>
          <div
            class="divider"
          />
          <div
            class="header"
          >
            Sorted Content
          </div>
          <a
            class="item"
            href="/sorted/"
            role="option"
          >
            Sorted by Title
          </a>
        </div>
      </div>
    </div>
  </div>,
  <div
    class="ui container"
    style="margin-top:5em"
  >
    <h1
      class="ui header"
    >
      React Kata
    </h1>
    <div
      class="ui vertical segment"
    />
  </div>,
  <div
    class="ui inverted vertical segment"
  >
    <div
      class="ui container"
    >
      <h2
        class="ui inverted header"
      >
        Summary of Instructions
      </h2>
      <ol
        class="ui inverted list"
        role="list"
      >
        <li
          class=""
          role="listitem"
        >
          Your software should read all data from the given CSV files in a meaningful structure.
        </li>
        <li
          class=""
          role="listitem"
        >
          Print out all books and magazines with all their details (with a meaningful output format).
        </li>
        <li
          class=""
          role="listitem"
        >
          Find a book or magazine by its \`isbn\`.
        </li>
        <li
          class=""
          role="listitem"
        >
          Find all books and magazines by their \`authors\`â€™ email..
        </li>
        <li
          class=""
          role="listitem"
        >
          Print out all books and magazines with all their details sorted by \`title\`. This sort should be done for books and magazines together.
        </li>
      </ol>
      <p>
        Please read full instructions on the 
        <a
          href="https://github.com/echocat/react-kata-1/blob/master/README.md"
          rel="noopener noreferrer"
          target="_blank"
        >
           README 
        </a>
         file
      </p>
    </div>
  </div>,
]
`;
